src/main/java/net/dries007/tfc/objects/Gem.java

```diff
@@ -23,11 +27,54 @@
 
     public enum Grade
     {
-        NORMAL,
-        FLAWED,
-        FLAWLESS,
-        CHIPPED,
-        EXQUISITE;
+        CHIPPED(16),
+        FLAWED(8),
+        NORMAL(4),
+        FLAWLESS(2),
+        EXQUISITE(1);
+
+        Grade(int dropWeight)
+        {
+            this.dropWeight = dropWeight;
+        }
+
+        // the probability of this gem grade dropping compared to the other grades. higher is more likely.
+        private int dropWeight;
+        private static int totalWeight = 31; // modify this if you change the individual drop weights
+
+        // chance of any gem dropping is determined by the general configs.
+        private static double dropChance = ConfigTFC.GENERAL.stoneGemDropChance;
+
+        /**
+         * Calculates the chances of a gem dropping from stone with dug
+         *
+         * @param random Random generator for rolling the odds
+         * @return null if no gem, otherwise the grade of the gem that should be dropped
+         */
```

Comment:
Wait, why did you want this Nullable here? This doesn't work at all.

Reply:
Because the method in question has an expected use case to return null.

Reply:
I ended up deciding it would be better if the type odds, grade odds, and drop odds were separated. That meant that this method no longer returns a null.